go/types/struct.go

```diff
@@ -16,46 +16,58 @@ import (
 )
 
 var EmptyStructType = MakeStructType("")
-var EmptyStruct = Struct{ValueSlice{}, EmptyStructType, &hash.Hash{}}
+var EmptyStruct = Struct{"", structValueFields{}, EmptyStructType, &hash.Hash{}}
 
 type StructData map[string]Value
 
+type structValueField struct {
+	name  string
+	value Value
+}
+
+type structValueFields []structValueField
+
+func (fs structValueFields) Len() int           { return len(fs) }
+func (fs structValueFields) Swap(i, j int)      { fs[i], fs[j] = fs[j], fs[i] }
+func (fs structValueFields) Less(i, j int) bool { return fs[i].name < fs[j].name }
+
 type Struct struct {
-	values []Value
+	name   string
+	fields structValueFields
```

Comment:
Not sure I follow? You would need to allocate the `[]string` for the names. Are you imagining that the encoder would inter that too?